fygen.py

```diff
@@ -252,6 +260,14 @@ def __init__(
     # Set to force sweep enable
     self.force_sweep_enable = False
 
+    # Detect model
+    if self.device_name is None:
+      model = self.get_model()
+      detected_device_name = model.split('-', 1)[0].lower()
```

Comment:
It seems good to check for '-' in the model name in case someone has a device with no '-' character.

e.g.

if '-' in model:
  detected_device_name = model.split('-', 1)[0].lower()
else:
  detected_device_name = model

Reply:
nevermind.  I see that the original code handles the case with no '-' just fine.

Reply:
Good point, though `.split` will leave the whole string intact if there is no boundary character in it, e.g.:

```
>>> "foo".split('-', 1)
['foo']
```